textile/chat-features.textile

```diff
@@ -9,5 +9,495 @@ jump_to:
+** @(CHA-RL2g)@ @[Testable]@ If all channel detachments complete successfully (all calls @detach()@ return successfully), then the room shall transition to the @DETACHED@ status.
+** @(CHA-RL2h)@ If during detachment, a channel fails to detach, different operations are performed based on the nature of the detach.
+*** @(CHA-RL2h1)@ @[Testable]@ If a channel enters the @FAILED@ state during detachment, then the room will transition to the @FAILED@ status. The detach operation continues until all channels are either @DETACHED@ or @FAILED@, with the operation throwing an @ErrorInfo@ with the feature-specific error code of the first feature to fail, using the underlying channel error as the @cause@. The same @ErrorInfo@ must accompany the @FAILED@ room status.
+*** @(CHA-RL2h2)@ @[Testable]@ If the room is already in a @FAILED@ status during the detach operation and another channel fails, the @FAILED@ status will not be emitted twice.
+*** @(CHA-RL2h3)@ @[Testable]@ If a channel enters another state during detachment (namely @ATTACHED@, which happens when detach times out). Then the detachment cycle will be retried after a short pause. The rooms status will not change during this process.
+* @(CHA-RL3)@ A room must be explicitly released via the @RELEASE@ operation.. This operation is not performed directly on a Room object by the client, but is described here.
+** @(CHA-RL3a)@ @[Testable]@ If the room is already in the @RELEASED@ status, this operation is no-op.
+** @(CHA-RL3b)@ @[Testable]@ If the room is in the @DETACHED@ status, then the room is immediately transitioned to @RELEASED@ and the operation succeeds.
+** @(CHA-RL3c)@ @[Testable]@ If the room is in the @RELEASING@ status, then the operation will return the result of the pending @release@ operation, or throw any exception that it throws.
+** @(CHA-RL3c)@ @[Testable]@ When the release operation commences, the room will transition into the @RELEASING@ status any any transient disconnect timeouts shall be cleared.
+** @(CHA-RL3d)@ @[Testable]@ All features channels are detached in sequence.
+** @(CHA-RL3e)@ @[Testable]@ If a channel is in the @FAILED@ state when it is time to detach, it shall be ignored.
+** @(CHA-RL3f)@ @[Testable]@ If a channel fails to detach, the channel detach sequence will be retried after a short wait.
+** @(CHA-RL3g)@ @[Testable]@ Once all channels have entered the @DETACHED@ state, then the room state is transitioned to @RELEASED@ and the operation completes.
+** @(CHA-RL3h)@ @[Testable]@ Upon operation completion, the underlying Realtime Channels are released from the core SDK prevent leakage.
+* @(CHA-RL5)@ A room must @RETRY@ whenever it enters the @SUSPENDED@ state.
+** @(CHA-RL5a)@ @[Testable]@ When entering a retry-loop, the room must first @DETACH@ all contributors underlying realtime channels.
```

Comment:
The term "retry loop" is used throughout CHA-RL5 but isn't defined. Is it synonymous with "RETRY operation"? The use of the word "loop" did help me understand a bit about the nature of the RETRY operation (i.e. that it repeats some behaviour in a loop until certain conditions are met), so perhaps we could either define "retry loop" or substitute it for "RETRY operation" but add some text about this operation’s looping nature to the (presumably) non-normative top level CHA-RL5a?

Reply:
Yes its synonymous with RETRY operation - I need to tidy up the language a bit at some point.

I suggest that CHA-RL5 becomes the non-normative requirement of the "retry loop" - as CHA-RL5 is supposed to basically be the definition of the "loop until condition met" behaviour.

Reply:
> Yes its synonymous with RETRY operation - I need to tidy up the language a bit at some point.

Do you intend to do it in this PR, or later?

> I suggest that CHA-RL5 becomes the non-normative requirement of the "retry loop" - as CHA-RL5 is supposed to basically be the definition of the "loop until condition met" behaviour.

I’m not sure what you mean by this, I’m afraid 🙂

Reply:
Fixed this - added some summary that hopefully makes more sense!